modules/analysis/src/service/load.rs

```diff
@@ -346,6 +347,267 @@ impl InnerService {
             .await
     }
 
+    #[instrument(skip(self, connection), err(level=Level::INFO))]
+    pub(crate) async fn load_latest_graphs_query<C: ConnectionTrait>(
+        &self,
+        connection: &C,
+        query: GraphQuery<'_>,
+    ) -> Result<Vec<(String, Arc<PackageGraph>)>, Error> {
+        let latest_sbom_ids = match query {
+            GraphQuery::Component(ComponentReference::Id(node_id)) => {
+                let sql = r#"
+                  SELECT distinct sbom_id
+                  FROM (
+                      SELECT sbom.sbom_id, sbom.published, product.id, cpe.id,
+                      RANK() OVER (PARTITION BY cpe.id ORDER BY sbom.published DESC) AS rank
+                      FROM sbom_node
+                      LEFT JOIN sbom ON sbom.sbom_id = sbom_node.sbom_id
+                      LEFT JOIN product_version on sbom.sbom_id = product_version.sbom_id
+                      LEFT JOIN product on product.id = product_version.product_id
+                      LEFT JOIN sbom_package_cpe_ref on sbom.sbom_id = sbom_package_cpe_ref.sbom_id
+                      LEFT JOIN cpe ON sbom_package_cpe_ref.cpe_id = cpe.id
+                      WHERE sbom_node.node_id = $1
+                  ) AS subquery
+                  WHERE rank = 1;                
+                "#;
+                let stmt = Statement::from_sql_and_values(
+                    DatabaseBackend::Postgres,
+                    sql,
+                    [node_id.into()],
+                );
+                let rows = connection.query_all(stmt).await?;
+                rows.into_iter()
+                    .filter_map(|row| {
+                        row.try_get_by_index::<Uuid>(0)
+                            .ok()
+                            .map(|sbom_id| sbom_id.to_string())
+                    })
+                    .collect::<Vec<String>>()
+            }
+            GraphQuery::Component(ComponentReference::Name(name)) => {
+                let sql = r#"
+                  SELECT distinct sbom_id
+                  FROM (
+                      SELECT sbom.sbom_id, sbom.published, product.id, cpe.id,
+                      RANK() OVER (PARTITION BY cpe.id ORDER BY sbom.published DESC) AS rank
+                      FROM sbom_node
+                      LEFT JOIN sbom ON sbom.sbom_id = sbom_node.sbom_id
+                      LEFT JOIN product_version on sbom.sbom_id = product_version.sbom_id
+                      LEFT JOIN product on product.id = product_version.product_id
+                      LEFT JOIN sbom_package_cpe_ref on sbom.sbom_id = sbom_package_cpe_ref.sbom_id
+                      LEFT JOIN cpe ON sbom_package_cpe_ref.cpe_id = cpe.id
+                      WHERE sbom_node.name = $1
```

Comment:
It doesn't seem like we need the product-related joins in any of these variants, do we? Tests pass without them.

```suggestion
                      SELECT sbom.sbom_id, sbom.published, cpe.id,
                      RANK() OVER (PARTITION BY cpe.id ORDER BY sbom.published DESC) AS rank
                      FROM sbom_node
                      LEFT JOIN sbom ON sbom.sbom_id = sbom_node.sbom_id
                      LEFT JOIN sbom_package_cpe_ref on sbom.sbom_id = sbom_package_cpe_ref.sbom_id
                      LEFT JOIN cpe ON sbom_package_cpe_ref.cpe_id = cpe.id
                      WHERE sbom_node.name = $1
```